packages/replay/src/upload.ts

```diff
@@ -8,278 +8,259 @@ import { Options, OriginalSourceEntry, RecordingMetadata, SourceMapEntry } from
+    await this.client.sendCommand("Internal.setRecordingMetadata", metadata);
+  }
 
-async function setRecordingMetadata(id: string, metadata: RecordingMetadata) {
-  if (!gClient) throw new Error("Protocol client is not initialized");
+  connectionProcessRecording(recordingId: string) {
+    if (!this.client) throw new Error("Protocol client is not initialized");
 
-  metadata.recordingData.id = id;
-  await gClient.sendCommand("Internal.setRecordingMetadata", metadata);
-}
+    this.client.sendCommand("Recording.processRecording", { recordingId });
+  }
 
-function connectionProcessRecording(recordingId: string) {
-  if (!gClient) throw new Error("Protocol client is not initialized");
+  async connectionWaitForProcessed(recordingId: string) {
+    if (!this.client) throw new Error("Protocol client is not initialized");
 
-  gClient.sendCommand("Recording.processRecording", { recordingId });
-}
+    const { sessionId } = await this.client.sendCommand<{ sessionId: string }>(
+      "Recording.createSession",
+      {
+        recordingId,
+      }
+    );
+    const waiter = defer();
 
-async function connectionWaitForProcessed(recordingId: string) {
-  if (!gClient) throw new Error("Protocol client is not initialized");
+    this.client.setEventListener("Recording.sessionError", ({ message }) =>
+      waiter.resolve(`session error ${sessionId}: ${message}`)
+    );
 
-  const { sessionId } = await gClient.sendCommand<{ sessionId: string }>(
-    "Recording.createSession",
-    {
-      recordingId,
-    }
-  );
-  const waiter = defer();
+    this.client.setEventListener("Session.unprocessedRegions", () => {});
 
-  gClient.setEventListener("Recording.sessionError", ({ message }) =>
-    waiter.resolve(`session error ${sessionId}: ${message}`)
-  );
+    this.client
+      .sendCommand("Session.ensureProcessed", { level: "basic" }, null, sessionId)
+      .then(() => waiter.resolve(null));
 
-  gClient.setEventListener("Session.unprocessedRegions", () => {});
+    const error = await waiter.promise;
+    return error;
+  }
 
-  gClient
-    .sendCommand("Session.ensureProcessed", { level: "basic" }, null, sessionId)
-    .then(() => waiter.resolve(null));
+  async connectionReportCrash(data: any) {
+    if (!this.client) throw new Error("Protocol client is not initialized");
 
-  const error = await waiter.promise;
-  return error;
-}
+    await this.client.sendCommand("Internal.reportCrash", { data });
+  }
 
-async function connectionReportCrash(data: any) {
-  if (!gClient) throw new Error("Protocol client is not initialized");
+  async connectionUploadRecording(recordingId: string, path: string) {
+    if (!this.client) throw new Error("Protocol client is not initialized");
 
-  await gClient.sendCommand("Internal.reportCrash", { data });
-}
+    const file = fs.createReadStream(path);
 
-// Granularity for splitting up a recording into chunks for uploading.
-const ChunkGranularity = 1024 * 1024;
+    file.on("error", e => {
```

Comment:
I think now that you're iterating with `for..await`, that will automatically catch and throw errors, so you shouldn't need this

Reply:
I wasn't sure if it would catch all instances but that makes sense.